clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -345,13 +400,61 @@ record_filter_t::parallel_shard_error(void *shard_data)
 bool
 record_filter_t::write_trace_entry(per_shard_t *shard, const trace_entry_t &entry)
 {
+    if (shard->output_entry_count == 0 && entry.type != TRACE_TYPE_HEADER) {
+        // When core-sharded with initially-idle cores we can start without a header.
+        // XXX i#6703: The scheduler should insert these headers for us.
```

Comment:
Is it too much more complex to add this support to the scheduler now? It seems to me that this is relevant for even a regular trace that is scheduled as core-sharded by the scheduler (a user may expect headers etc on each core even for that case)

Reply:
This is an issue that could affect any tool or simulator that cares about headers (augmented this comment to reflect that): but OTOH we never hit this until now when I added this 1-to-4 test.  Certainly it is cleaner in the scheduler, but given other project priorities we want to make progress on the main use case here, as discussed offline.